website/django/data/tests.py

```diff
@@ -82,6 +93,186 @@ def test_autocomplete_bic(self):
+        response = variant_search(req)
+        self.assertJSONEqual(response.content, ErrorMessages['start'])
+
+        request.start = 14589
+        req = self.factory.post("/data/ga4gh/variants/search", json.dumps(json_format._MessageToJsonObject(request, False)),
+                                content_type="application/json")
+        response = variant_search(req)
+        self.assertJSONEqual(response.content, ErrorMessages['end'])
+
+    def test_offset_calculator(self):
+        start = 12500
+        end = 13000
+        s1, e1 = get_offset(start, end) #Ending offset is the same, the query will be exclusive, and therefore query will satisfy type, that is inclusive.
+        self.assertEquals(s1, start+1)
+        self.assertEquals(e1, end+1)
+        opperlen = ['a','b']
+        s2, e2 = get_offset(start, end, opperlen)
+        self.assertEquals(s2, start - 1)
+        self.assertEquals(e2, s2+2)
+
+    def test_pagging_token(self):
+        request0 = self.factory.post(
+            "/data/ga4gh/variants/search", json.dumps(
+                {"end": 45158029, "referenceName": "chr13", "variantSetId": "brca-hg37", "start": 20425158,
+                 "pageSize": 5}), content_type="application/json")
+        response = variant_search(request0)
+        self.assertEqual(json.loads(response.content)["nextPageToken"], "1" )
+
+        request0 = self.factory.post(
+            "/data/ga4gh/variants/search", json.dumps(
+                {"end": 51425158029, "referenceName": "chr17", "variantSetId": "brca-hg37", "start": 41245664,
+                 "pageSize": 5, "pageToken": "3"}), content_type="application/json", )
+        response = variant_search(request0)
+        self.assertEqual(json.loads(response.content)["nextPageToken"], "4")
+        self.assertEquals(len(json.loads(response.content)["variants"]), 5)
+
+    def test_get_variant_response(self):
+        request = self.factory.get("/data/ga4gh/variants/hg37-1")
+        response = get_var_by_id(request, "hg37-19")
+        jsonresp = json.loads(response.content)
+        self.assertIsNotNone(jsonresp["referenceName"], True)
+        self.assertIsNotNone(jsonresp["start"], True)
+        self.assertTrue(jsonresp["start"] == 41275598)
+        self.assertTrue(jsonresp["end"] == 41275599)
+
+    def test_brca_to_ga4gh_variantSets_status_code(self):
+        request0 = self.factory.post(
+            "/data/ga4gh/variants/search", json.dumps({"datasetId": "brca-exchange"}), content_type="application/json")
+        response = get_variantSet(request0)
+        self.assertEqual(response.status_code, 200)
+
+    def test_brca_to_ga4gh_variantSets_metafieldNums(self):
+        request0 = self.factory.post(
+            "/data/ga4gh/variantsets/search", json.dumps({"datasetId": "brca-exchange"}), content_type="application/json")
+        response = get_variantSet(request0)
+        self.assertEqual(len(json.loads(response.content)["variantSets"][0]["metadata"]), 83)
```

Comment:
Check metadata is well formed. Note: when you're looping over something to test the inner elements, make sure the list you are iterating over is not empty first.